Add macaddr encoding and decoding

kaol opened the pull request.

nikita-volkov commented:
Great! Thanks!

Can you add tests? Preferably the following:

- A roundtrip, which ensures that what is decoded from the DB and encoded back is equal to the original.
- A roundtrip, which generates random values on the Haskell side, trims the ignored bits, encodes and decodes them and ensures that the trimmed value is produced.

kaol commented:
I added a test.

I switched from a `Word64` representation to a 6-tuple of `Word8` values after all. I'm not familiar enough with `QuickCheck` to easily come up with the test you described with discarding upmost bits from a `Word64` and changing the representation sidesteps the whole issue.

Also, this is closer to what `network-info` uses for a representation and `ip` has easy conversions from/to octets as well.

nikita-volkov commented:
Thanks! Merging and releasing.
